### PR TITLE
cleanup remote dut initialization

### PR DESCRIPTION
**Pull request by jupe**
just cleanup a bit remote dut initialization phase

### CODE REVIEW

File: mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py [hunk 1]
```diff
@@ -73,9 +73,8 @@ def __remote_init(self):
 
         # Remote DUT connection, flashing and reset...
         try:
-            self.__remote_connect(baudrate=self.baudrate)
```
Comment (jupe) [inline mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py]: this is just waste of time :)

File: mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py [hunk 2]
```diff
@@ -73,9 +73,8 @@ def __remote_init(self):
...
-            self.__remote_flashing(self.image_path)
             self.__remote_disconnect()
```
Comment (jonikula) [inline mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py]: Does this disconnect need to be here, since we don't connect any more?

Comment (jupe) [inline mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py]: if check first if there was connection - if was then disconnect. this is neede to select right serialport configuration when connecting again..

Comment (jonikula) [inline mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py]: So the __remote_disconnect actually checks if there is a connection? If so, this is ok.

Comment (jonikula) [inline mbed_host_tests/host_tests_conn_proxy/conn_primitive_remote.py]: Yea, it seems to have a check for this. lgtm, though I have no experience with htrun so I'll leave it for someone else to approve this.

**Pull request merged**